src/cli.rs

```diff
@@ -8,9 +8,31 @@ use crate::{
 use anyhow::Result;
 use clap::{Args, CommandFactory, Parser};
 use clap_complete::Shell;
-use reqwest::Client;
+use reqwest::{Client, Error};
 use tracing::{error, info};
 
+fn create_reqwest_client() -> Result<Client, Error> {
+    // fetch env variable
+    let github_token = match std::env::var("GITHUB_TOKEN") {
```

Comment:
Instead of just doing so keep it as Result<String> and then
```rs
if let Ok(v) = github_token {
// le code 
}
```

Reply:
As far as I understand rust, the purpose of current implementation is to have environment value or to have an empty string if not present.

And based on that make further logic for header values.

If it's not valid can you please explain in a bit of detail.

Reply:
Never mind I think I got you. Let me give that a try.

Reply:
Pushed new changes, please review.